source/joker_effects.c

```diff
@@ -212,69 +238,75 @@ static JokerEffect joker_stencil_effect(Joker *joker, Card *scored_card) {
 }
 
 #define MISPRINT_MAX_MULT 23
-static JokerEffect misprint_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect misprint_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
+
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
 
     effect.mult = random() % (MISPRINT_MAX_MULT + 1);
 
     return effect;
 }
 
-static JokerEffect walkie_talkie_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect walkie_talkie_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card == NULL)
-        return effect;
 
-    if (scored_card->rank == TEN || scored_card->rank == FOUR) {
-            effect.chips = 10;
-            effect.mult = 4;
-    }
+    SCORE_ON_CARD_SCORED_ONLY(scored_card, scored_when, effect)
 
+    if (scored_card && (scored_card->rank == TEN || scored_card->rank == FOUR)) {
+        effect.chips = 10;
+        effect.mult = 4;
+    }
+    
     return effect;
 }
 
-static JokerEffect fibonnaci_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect fibonnaci_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card == NULL)
-        return effect;
+
+    SCORE_ON_CARD_SCORED_ONLY(scored_card, scored_when, effect)
```

Comment:
```suggestion
    SCORE_ON_EVENT_ONLY_WITH_CARD(JOKER_CALLBACK_ON_CARD_SCORED, scored_card, scored_when, effect)
```
Batch - Consolidate joker event restriction macros